### ISSUE TITLE
Support for multiple toolsets

### ISSUE THREAD

**Issue by mthalman** [dotnet/arcade#2999]
We'd like to get our Docker-related repos onboarded to Arcade to take advantage of having a central definition of common infrastructure files that are automatically propagated amongst all our repos.  The problem is that the Arcade SDK currently defines a toolset that's all about building source code.  In the Docker world, we're not building source code but rather Docker images.  So all of the existing tools in Arcade SDK are irrelevant to what we need.

The request is for Arcade to support multiple toolsets so that there is strict isolation between them and the files that are propagated.

There are several reasons why it's important to have this isolation and avoid mixing the current toolset with our Docker infrastructure files:
* We want our Docker repos to contain Docker-related infrastructure.  Having irrelevant files like MSBuild-related scripts just pollutes the repo and leads to confusion of what is relevant.
* When changes to eng\common are propagated through PRs, it triggers builds to be run.  Our Docker builds make extensive use of a variety of build agents, many of which are non-Hosted (custom) and we've got limited resources as a result.  We do not want to be wasting these resources for changes to files that are irrelevant to what our infrastructure requires.
* Then there's just the matter of keeping the repo history clean.  We'd like to avoid having changes to files unrelated to our infrastructure polluting the git history.

**Comment by tmat** [comment]
Seems similar request to https://github.com/dotnet/arcade/issues/2803

**Comment by markwilkie** [comment]
Yup - likely the same implementation will fix both scenarios which is good.

@mthalman - Is this work something you and @MichaelSimons  think is something your team might do or do you plan to wait for dnceng?

**Comment by mthalman** [comment]
@markwilkie - Since this isn't something we're blocked on, I think this is something that we'd just wait for dnceng to work on.

**Comment by markwilkie** [comment]
Moving to our docker epic (business priority)
